.github/actions/javascript/checkAndroidStatus/checkAndroidStatus.ts

```diff
@@ -0,0 +1,98 @@
+import * as core from '@actions/core';
+import {google} from 'googleapis';
+import CONST from '@github/libs/CONST';
+import GithubUtils from '@github/libs/GithubUtils';
+
+const PACKAGE_NAME = core.getInput('PACKAGE_NAME', {required: true});
+const GOOGLE_KEY_FILE = core.getInput('GOOGLE_KEY_FILE', {required: true});
+const HALTED_STATUS = 'halted';
+
+async function checkAndroidStatus() {
+    const auth = new google.auth.GoogleAuth({
+        keyFile: GOOGLE_KEY_FILE,
+        scopes: ['https://www.googleapis.com/auth/androidpublisher'],
+    });
+
+    const androidApi = google.androidpublisher({
+        version: 'v3',
+        auth,
+    });
+
+    try {
+        // Insert an edit to get an edit ID
+        const editResponse = await androidApi.edits.insert({
+            packageName: PACKAGE_NAME,
+        });
+        const editId = editResponse.data.id ?? 'undefined';
+
+        // Get the production track status
+        const trackResponse = await androidApi.edits.tracks.get({
+            packageName: PACKAGE_NAME,
+            editId,
+            track: 'production',
+        });
+
+        const status = trackResponse.data.releases?.[0]?.status ?? 'undefined';
+        console.log('Track status:', status);
+
+        // Check if the status is halted
+        const HALTED = status === HALTED_STATUS;
+        core.setOutput('HALTED', HALTED);
+    } catch (error) {
+        console.error('Error checking track status:', error);
+        process.exit(1);
+    }
+}
+
+async function getLatestReleaseDate() {
+    const {data} = await GithubUtils.octokit.repos.getLatestRelease({
+        owner: CONST.GITHUB_OWNER,
+        repo: CONST.APP_REPO,
+    });
+
+    const releaseDate = data.published_at?.split('T')[0];
```

Comment:
Do you know if this is millis or formatted date? It doesn't matter but a comment here with the format might be useful in the future for people editing the file

Reply:
Maybe ignore this actually, I see below it's a string that we pass into DATE